internal/pkg/agent/cmd/container.go

```diff
@@ -962,3 +963,13 @@ func isContainer(detail component.PlatformDetail) component.PlatformDetail {
 	detail.OS = component.Container
 	return detail
 }
+
+// isStatePathTooLong returns true if the final Unix socket path
+// will be too long and needs to be shortened.
+//
+// Source: https://www.man7.org/linux/man-pages/man7/unix.7.html
+func isStatePathTooLong(statePath string) bool {
+	// This replicates the logic from `setPaths`
+	socketPath := filepath.Join(statePath, "data", paths.ControlSocketName)
+	return len(socketPath) > 107
```

Comment:
Why 107? We are using 104 everywhere else.

Reply:
After reading https://www.man7.org/linux/man-pages/man7/unix.7.html, 107 seemed the most appropriate. It was also the longest I could get without the socket failing to be created.

I can chance to 104.

Reply:
It got removed with my lats few commits